### PR TITLE
feat(api): method signature change for update/create product methods

### PR DESCRIPTION
**Pull request by jimmythomson**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Gradient Edge commercetools utils
 
+# ************************
+# DO NOT USE THIS LIBRARY #
```
Comment (jameswiltshire) [inline README.md:4]: "BETA - Not recommended for production use"  ? :)

File: src/lib/api/CommercetoolsApi.ts [hunk 2]
```diff
@@ -361,6 +379,10 @@ export class CommercetoolsApi {
     if (process?.release?.name) {
       headers['User-Agent'] = this.userAgent
     }
+    if (typeof options.correlationId === 'string') {
```
Comment (jameswiltshire) [inline src/lib/api/CommercetoolsApi.ts:382]: Do you need to check for a non empty/null string here too?

**Review by jameswiltshire — APPROVED** [review]

File: src/lib/api/CommercetoolsApi.ts (hunk 2 above)
Comment (jimmythomson) [reply src/lib/api/CommercetoolsApi.ts:382]: Nope - if it's null then typeof is 'object', and undefined is typeof of 'undefined'

Comment (jimmythomson) [reply src/lib/api/CommercetoolsApi.ts:382]: Ah, sorry, mis-read the question. Yep, maybe an empty string check. I'll add that in.

File: README.md (hunk 1 above)
Comment (jimmythomson) [reply README.md:4]: Will update.

**Pull request merged**